.github/workflows/testing.yml

```diff
@@ -0,0 +1,45 @@
+name: Testing workflow
+
+on: [push]
+
+jobs:
+  style:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: "Install style checker clang-format"
+        run: sudo apt update && sudo apt install clang
+      - name: "Run style checker clang-format"
+        run: clang-format -i src/*.hpp src/*.cpp tests/*.cpp
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: "Install necessary C++ dependencies"
+        run: sudo apt update && sudo apt install cmake gcc libeigen3-dev libyaml-cpp-dev libboost-all-dev
+      - name: "Create build folder"
+        run: mkdir build
+      - name: "Run cmake"
+        working-directory: ./build
+        run: cmake ..
+      - name: "Build make"
+        working-directory: ./build
+        run: make
+  test:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: "Install necessary C++ dependencies"
+        run: sudo apt update && sudo apt install cmake gcc libeigen3-dev libyaml-cpp-dev libboost-all-dev
+      - name: "Create build folder"
+        run: mkdir build
+      - name: "Run cmake"  
+        working-directory: ./build
+        run: cmake ..
+      - name: "Make"
+        working-directory: ./build
+        run: make
```

Comment:
Here, you build again. For larger projects, this might not be what we want. We could instead reuse the build from the `build` job through an artifact.